src/lib/y2users/config_manager.rb

```diff
@@ -81,5 +81,19 @@ def system(reader: nil, force_read: false)
 
       config
     end
+
+    # Gets target modification used to store configuration before it is applied
+    # @return [Config,nil] returns nil if no target is yet initialized
+    def target
+      config(:target)
+    end
+
+    # Sets target modification used to store configuration before it is applied
+    # @param [Config] config Note that config should not be frozen as it is expected to be modified
+    # @note in various scenarious target does not need to be exact target state, but just
+    # interstate that is then merged with real system state
+    def target=(config)
+      register(config, as: :target)
```

Comment:
Should we check that it is a Y2Users::Config before registering it?

Reply:
well, we do not do it also in register...if we do it in register, we get it for free here. So to be consistent just do what register do.

Reply:
Right!